### PR TITLE
Daemonize etcd

### PR DESCRIPTION
**Pull request by gaurav36**
Do not kill etcd daemon upon issuing ^c (SIGINT) in glusterd daemon
    
    Currently when etcd daemon start upon starting of glusterd2 then upon
    performing ^C (SIGINT) its killing etcd daemon.
    
    With this patch it will set child process id to different from main
    process id group and, etcd daemon will not be kill upon performing
    SIGINT (^C) of main process.
    
 
    
    Issue #66
    
    Signed-off-by: Gaurav Kumar Garg <garg.gaurav52@gmail.com>

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -266,3 +269,62 @@ func isBrickPathAlreadyInUse(brickPath string) bool {
...
+			"path": dir,
+		}).Fatal("directory path is not a writable")
+	}
+	// defer happens in LIFO
+	defer syscall.Unlink(t.Name())
+	defer t.Close()
+}
+
+// Function to check whether the process with given pid exist or not in the system
+func CheckPidExist(pid int) bool {
```
Comment (atinmu) [inline utils/utils.go]: Isn't it better to name it as CheckProcessExist?

File: etcdmgmt/etcd-mgmt.go [hunk 2]
```diff
@@ -39,43 +88,70 @@ func StartEtcd() (*exec.Cmd, error) {
 		"-initial-advertise-peer-urls", initialAdvPeerUrls,
 		"--initial-cluster", "default="+listenPeerUrls)
 
+	// Don't kill chlid process (etcd) upon ^C (SIGINT) of main glusterd process
+	etcdCmd.SysProcAttr = &syscall.SysProcAttr{
```
Comment (atinmu) [inline etcdmgmt/etcd-mgmt.go]: Please add a TODO to remove syscall usage

File: context/context.go [hunk 3]
```diff
@@ -29,7 +31,7 @@ var (
 	Rest      *rest.GDRest
 	TxnFw     *transaction.GDTxnFw
 	OpVersion int
-	EtcdCtx   *exec.Cmd
+	EtcdCtx   *os.Process
```
Comment (atinmu) [inline context/context.go]: Can we rename this to ETCDProcessCtx?

File: etcdmgmt/etcd-mgmt.go [hunk 4]
```diff
@@ -6,16 +6,65 @@ import (
...
+		if result.Health == "true" {
+			timer.Stop()
+			break
+		}
+	}
+	return true
+}
+
+var (
+	etcdPidDir  = "/var/run/gluster/"
```
Comment (kshlm) [inline etcdmgmt/etcd-mgmt.go]: This cannot be hardcoded. If you are going to store the etcdPid in GD2's workdir, use config.WorkDir. Don't hardcode paths like this.

**Pull request merged**